src/Homely.AspNetCore.Mvc.Helpers/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -72,7 +73,7 @@ public static class IServiceCollectionExtensions
                                                                   Func<ApiDescription, string> operationIdSelector = null)
         {
             services.AddControllers()
-                    .AddAHomeController(services, banner)
+                    .AddAHomeController(services, banner, Assembly.GetCallingAssembly())

```

Comment:
i don't get it. the `AddAHomeController` method does this same code, when null anyway: https://github.com/Homely/Homely.AspNetCore.Mvc.Helpers/blob/0ea02920e5ca67728199d3e2541b281d9ca197d6/src/Homely.AspNetCore.Mvc.Helpers/Extensions/IMvcBuilderExtensions.cs#L33

So, not sure how this change does anything?

Reply:
![](https://media.giphy.com/media/cRKRlRPewM9NQCXRvf/giphy.gif)

inside the `AddAHomeController` we display the name of _some_ assembly that is provided OR the current assembly (which is this library).

```
Some Test Api
Name: TestWebApplication   <--- HERE
...
```

so in this Default (aka. simple way of doing this with 'default' settings) method, I'm saying: use the Assembly which _called_ this method.

this is usually called in `startup.cs` from the application _host_ (webhost or console host).

so this 'figures' that out.

and then the correct name is displayed (as above).

Reply:
got it. in that case then, why make it an optional param? ie when would it _ever_ be useful to have:

```
Name: Homely.AspNetCore.Mvc.Helpers
```

Reply:
actually hang on `Assembly.GetCallingAssembly(); ` should be using the CALLING assembly, right? e.g the host?

>  OR the current assembly (which is this library).
so it should never use the 'current' assembly?

Reply:
- `Current` assembly is a fall back in case nothing is provided.
- CALLING can be anything you want. you can give the param _any_ Assembly. I'm not going to assume you want to use the host assembly name
- `AddDefaultWebApiSettings` is just a helper wrapper method with default settings. So for this _opinionated method_ I've defaulted to whatever called this method. I'm _assuming_ that would be a host.

Reply:
If you want, you don't need to use the `AddDefaultWebApiSettings` method(s). they are just helpers.

you can manually call whatever you want. for example, you actually might not want to have a HomeController (or _our_ HomeController).

If you do manually want to wire up our HomeController, then you can provide some Assembly and again, you decide what that assembly is/would be.

Reply:
Discussed further in Slack and the issue/confusion was that i was saying the wrong thing (above).

- `AddAHomeController` calls Calling. ✅ 
- `AddDefaultWebApiSettings` then calls `AddAHomeController` which then says says "I called myself".

so `AddDefaultWebApiSettings` needs to also call Calling so downstream doesn't resolve to itself.